### PR TITLE
Fix Issue 21283 - [C++] Wrong mangling for ref of parameter pack

### PR DESCRIPTION
**Pull request by BorisCarvajal**

### CODE REVIEW

**Comment by BorisCarvajal** [comment]
Where is dlang-bot?

File: src/dmd/cppmangle.d [hunk 1]
```diff
@@ -1468,6 +1472,28 @@ private final class CppMangleVisitor : Visitor
...
+     */
+    private void checkParamPack(Type t, TemplateParameters* params)
+    {
+        TypeIdentifier ti = void;
+        if (t.isTypeIdentifier())
+            ti = cast(TypeIdentifier)t;
+        else if (t.isTypeReference() && t.nextOf().isTypeIdentifier())
+            ti = cast(TypeIdentifier)t.nextOf();
+        else
+            return;
```
Comment (thewilsonator) [inline src/dmd/cppmangle.d:1490]: Please do 
```d
if (t.isTypeReference())
    t = t.nextOf();
auto ti = t.isTypeIdentifier();
if (!ti)
    return;
...
```

File: src/dmd/cppmangle.d [hunk 2]
```diff
@@ -1468,6 +1472,28 @@ private final class CppMangleVisitor : Visitor
...
     }
 
+    /**
+     * Write `Dp` (C++11 function parameter pack prefix) if 't' is a TemplateSequenceParameter (T...).
+     *
+     * Params:
+     *   t      = Parameter type
+     *   params = Template parameters of the function
+     */
+    private void checkParamPack(Type t, TemplateParameters* params)
```
Comment (thewilsonator) [inline src/dmd/cppmangle.d:1482]: I don't really like `check` since that signals checking for errors to me. I'm not sure what term would be better though.

Comment (Geod24) [reply src/dmd/cppmangle.d:1482]: Usually I would use `handle` in that situation.

Comment (BorisCarvajal) [reply src/dmd/cppmangle.d]: I changed it to `handle`.

**Review by Geod24 — APPROVED** [review]

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**